### DISCUSSION TITLE
Using the same Collection in multiple components

### DISCUSSION THREAD

**Discussion by Joonaspraks** [adobe/react-spectrum#4722] [General]
I am constructing a multiselect in react-aria. I use the same Collection to both display Options in the ListBox and display the selected Options in a TagGroup. Because I am using the  same Collection in both components I get overlapping focus events. For example, hovering on a selected ListBox Option sets focus on the matching Tag.

What are the approaches for this problem? I could override the keys for the TagGroup, but then I lose the out-of-the-box support for the Tag's onRemove capabilities.

There is a similar discussion here: https://github.com/adobe/react-spectrum/issues/3785

**Comment by LFDanLu** [comment]
Thinking out loud, perhaps we could look into adding a unique identifier to each SelectionManager that would then be added a option's generated data-key along with its key [here](https://github.com/adobe/react-spectrum/blob/2b6302bbec3ab48c9efb22b88d521a37227385ec/packages/%40react-aria/selection/src/useSelectableItem.ts#L268). That way the [query selector used for focus](https://github.com/adobe/react-spectrum/blob/2b6302bbec3ab48c9efb22b88d521a37227385ec/packages/%40react-aria/selection/src/useSelectableCollection.ts#L319) would be scoped to the selection manager + collection pair rather than to the collection only. 

EDIT: 
For clarification, is your TagGroup and ListBox within the same scrollRef (scrolling container)? If not, then I'm surprised that this is an issue since the code that controls focus usually only is triggered if it the overall collection has focus within it
